File: 2021/2021W24/preppindataW24.R
library(readxl)
library(dplyr)
library(lubridate)
library(fuzzyjoin)

final <- read_excel(path = "F:/Data/Absenteeism Scaffold.xlsx", sheet = 'Reasons') %>%
  mutate('End Date' = `Start Date` %m+% days(`Days Off`)-1) %>%
  fuzzy_full_join(., as.data.frame(seq(as.Date("2021/4/1"), as.Date("2021/5/31"),by="day")) %>%
                    `colnames<-`(c('Date')),
                  by=c('Start Date'='Date', 'End Date'='Date'), match_fun=list(`<=`,`>=`)) %>%
  group_by(`Date`) %>%
  summarise('Number of people off each day' = sum(!is.na(`Name`)), .groups='drop')

ans1 <- final %>%
  filter(`Number of people off each day` == max(`Number of people off each day`)) %>%
  .[['Date']]
  
ans2 <- final %>%
  filter(`Number of people off each day` == 0) %>%
  nrow(.)

View(final)

File: 2021/2021W24/PreppindataW24.py
import pandas as pd
from datetime import timedelta

def timeoff(df, d):
    return len(df[(df['Start Date']<=d) & (d<=df['End Date'])])

df = pd.read_excel(pd.ExcelFile("F:/Data/Absenteeism Scaffold.xlsx"),'Reasons')
df['End Date'] = df.apply(lambda x: x['Start Date']+timedelta(days=x['Days Off']-1), axis=1)

final = pd.DataFrame(pd.date_range(start='2021/4/1', end='2021/5/31'), columns=['Date'])
final['Number of people off each day'] = final['Date'].apply(lambda x: timeoff(df,x))

ans1 = final[final['Number of people off each day']==max(final['Number of people off each day'])]['Date'].copy().iloc[0]
ans2 = len(final[final['Number of people off each day']==0])
